src/qml/LayerTreeItemProperties.qml

```diff
@@ -14,28 +14,33 @@ Popup {
   property var index
 
   property bool zoomToLayerButtonVisible: false
+
   property bool trackingButtonVisible: false
   property var trackingButtonText
 
-  property alias itemVisible: itemVisibleCheckBox.checked
+  property bool expandCheckBoxVisible: false
+
 
   width: mainWindow.width - 20
   x: (parent.width - width) / 2
   y: (parent.height - height) / 2
   padding: 0
 
   onIndexChanged: {
-    itemVisible = layerTree.data(index, FlatLayerTreeModel.Visible)
     title = layerTree.data(index, Qt.DisplayName)
+
+    itemVisibleCheckBox.checked = layerTree.data(index, FlatLayerTreeModel.Visible);
+
+    expandCheckBoxVisible = layerTree.data(index, FlatLayerTreeModel.HasChildren)
+    expandCheckBox.text = layerTree.data( index, FlatLayerTreeModel.Type ) === 'group' ? qsTr('Expand group') : qsTr('Show legend')
```

Comment:
```suggestion
    expandCheckBox.text = layerTree.data( index, FlatLayerTreeModel.Type ) === 'group' ? qsTr('Expand group') : qsTr('Show layer legend')
```
I know, that we decided this in hive. Still while using it it seems strange. You make a double tip an a legend item and there is the option "Show legend" - I can imagine that users think it's the whole legend we are talking about.

Reply:
Rule based renderer can create a tree with subfolders of its own. Show layer legend would be odd for these items.

What about show legend item children?

Reply:
In the chat yesterday we decided against the children. But I don't mind that much.
Or maybe keep it simple with "Expand legend item"?

Reply:
+1 to expand legend item